character_map_tester.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright:  Copyright Dimakopoulos Theodoros 2019
+ * License:    $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:    Dimakopoulos Theodoros
+ */
+#pragma once
+#include "character_map.h"
+
+#include <stdio.h>
+
+// highly technical function
+// to understand the output, check out CharacterMap.h
+void ch_map_test() {

```

Comment:
Ok, so you want an automated way to do testing. Printing is not automated because person has to look at the output and verify that it is correct ;) You want a way that a computer checks that the code is correct. That can be done with `assert()`. Do a quick google search if you don't know it, it's very simple.
To use it effectively, you try to find things that _must_ be true when your code runs, e.g. `ch_map['0'] & CM_DECIMAL`, so you `assert` it.